src/scrapers/leumi.js

```diff
@@ -119,6 +103,75 @@ async function fetchTransactionsForAccount(page, startDate) {
     }
   }
 
+  return txns;
+}
+
+async function extractPendingTransactionsFromPage(page) {
+  const txns = [];
+
+  const tdsValues = await page.$$eval('#WorkSpaceBox #trTodayActivityNapaTableUpper tr td', (tds) => {
+    return tds.map(td =>
+      ({
+        classList: td.getAttribute('class'),
+        innerText: td.innerText,
+      }));
+  });
+
+  for (const element of tdsValues) {
+    if (element.classList.includes('Colume1Width')) {
+      const newTransaction = { status: 'pending' };
+      newTransaction.date = (element.innerText || '').trim();
+      txns.push(newTransaction);
+    } else if (element.classList.includes('Colume2Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.description = element.innerText;
+      txns.push(changedTransaction);
+    } else if (element.classList.includes('Colume3Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.reference = element.innerText;
+      txns.push(changedTransaction);
+    } else if (element.classList.includes('Colume4Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.debit = element.innerText;
+      txns.push(changedTransaction);
+    } else if (element.classList.includes('Colume5Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.credit = element.innerText;
+      txns.push(changedTransaction);
+    } else if (element.classList.includes('Colume6Width')) {
+      const changedTransaction = txns.pop();
+      changedTransaction.balance = element.innerText;
+      txns.push(changedTransaction);
+    }
+  }
+
+  return txns;
+}
+
+async function fetchTransactionsForAccount(page, startDate) {
+  await dropdownSelect(page, 'select#ddlTransactionPeriod', '004');
+  await waitUntilElementFound(page, 'select#ddlTransactionPeriod');
+  await fillInput(
+    page,
+    'input#dtFromDate_textBox',
+    startDate.format('DD/MM/YY'),
```

Comment:
@esakal not new code, but would be a good time to extract the date format into a constant

Reply:
I'm not sure, I'm a fan of constants where the value is used in multiple places. The date format value is specific to this function logic so I don't think we should move it outside . wdyt?

Reply:
@esakal I usually try to extract important variables into constants even if they're only used once, so they appear loud and clear at the top of files. Again, see your point, so not going to insist over that either.